internal/rancher-api/api.go

```diff
@@ -80,17 +68,44 @@ func (r Rancher) CreateOperator(pipelineId string, input lib.Operator, outputTop
 	}
 	request := gorequest.New().SetBasicAuth(r.accessKey, r.secretKey)
 
-	reqBody := &Request{
-		Type:          "service",
-		Name:          r.GetOperatorName(pipelineId, input),
-		StackId:       r.stackId,
-		Scale:         1,
-		StartOnCreate: true,
-		LaunchConfig: LaunchConfig{
-			ImageUuid:   "docker:" + input.ImageId,
-			Environment: env,
-			Labels:      labels,
-		},
+	reqBody := &Request{}
+	if useMetrics {
+		env["METRICS_URL"] = metricsConfig.Url
+		env["METRICS_USER"] = metricsConfig.Username
+		env["METRICS_PASSWORD"] = metricsConfig.Password
+		env["METRICS_INTERVAL"] = metricsConfig.Interval
+
+		reqBody = &Request{
+			Type:          "service",
+			Name:          r.GetOperatorName(pipelineId, input),
+			StackId:       r.stackId,
+			Scale:         1,
+			StartOnCreate: true,
+			LaunchConfig: LaunchConfig{
+				ImageUuid:   "docker:" + input.ImageId,
+				Environment: env,
+				Labels:      labels,
+				Command: []string{
+					"java",
+					"-javaagent:jmxtrans-agent.jar=" + metricsConfig.XmlUrl,
```

Comment:
Requires operators to download jmxtrans-agent.jar at build. Add this line to Dockerfiles:
`ADD https://github.com/jmxtrans/jmxtrans-agent/releases/download/jmxtrans-agent-1.2.6/jmxtrans-agent-1.2.6.jar jmxtrans-agent.jar
`